components/datetime/src/raw/neo.rs

```diff
@@ -186,7 +186,11 @@ impl DatePatternSelectionData {
             year_style,
             input
                 .year
-                .map(|y| y.year_ambiguity())
+                .map(|y| {
+                    y.era()
+                        .map(|e| e.ambiguity)
+                        .unwrap_or(YearAmbiguity::EraRequired)
+                })
```

Comment:
Thought: this looks suspicious but it is the same as the previous logic

Reply:
_big shrug_

Reply:
yeah this code makes is obvious what happens here, maybe this function can be cleaned up more